lib/bugsnag.rb

```diff
@@ -212,27 +209,40 @@ def leave_breadcrumb(name, meta_data={}, type=Bugsnag::Breadcrumbs::MANUAL_BREAD
       validator.validate(breadcrumb)
 
       # Skip if it's already invalid
-      unless breadcrumb.ignore?
-        # Run callbacks
-        configuration.before_breadcrumb_callbacks.each do |c|
-          c.arity > 0 ? c.call(breadcrumb) : c.call
-          break if breadcrumb.ignore?
-        end
+      return if breadcrumb.ignore?
 
-        # Return early if ignored
-        return if breadcrumb.ignore?
+      # Run callbacks
+      configuration.before_breadcrumb_callbacks.each do |c|
+        c.arity > 0 ? c.call(breadcrumb) : c.call
+        break if breadcrumb.ignore?
+      end
+
+      # Return early if ignored
+      return if breadcrumb.ignore?
 
-        # Validate again in case of callback alteration
-        validator.validate(breadcrumb)
+      # Validate again in case of callback alteration
+      validator.validate(breadcrumb)
 
-        # Add to breadcrumbs buffer if still valid
-        configuration.breadcrumbs << breadcrumb unless breadcrumb.ignore?
+      # Add to breadcrumbs buffer if still valid
+      configuration.breadcrumbs << breadcrumb unless breadcrumb.ignore?
+    end
+
+    ##
+    # Returns the client's Cleaner object, or creates one if not yet created.
+    #
+    # @api private
+    #
+    # @return [Cleaner]
+    def cleaner
+      @cleaner = nil unless defined?(@cleaner)
+      @cleaner || LOCK.synchronize do
+        @cleaner ||= Bugsnag::Cleaner.new(configuration)
       end
```

Comment:
```suggestion
      @cleaner ||= LOCK.synchronize do
        Bugsnag::Cleaner.new(configuration)
      end
```

[very tiny nit] You can move the `=` up to make or/assign blocks easier to read.

<details>
<summary>test case</summary>

```rb
class Foo
  LOCK = Mutex.new

  def self.foo
    @foo ||= LOCK.synchronize do
      puts "Assigning value"
      5
    end
  end
end

puts Foo.foo
puts Foo.foo
```
prints:
```
Assigning value
5
5
```

</details>

Reply:
I think the current way is actually required to be thread safe, which doesn't _really_ matter in this case but it's consistent with the `configuration` and `session_tracker` methods

If multiple threads manage to call this at the exact same time, they can both get into the `synchronize` block and set `@cleaner` to a new instance, so I think it's necessary to do this even if it is a bit ugly :p

<details>
<summary>test case</summary>

```ruby
class Foo
  LOCK = Mutex.new

  def self.foo
    @foo ||= LOCK.synchronize do
      puts "Assigning value in #{Thread.current}"
      rand(100)
    end
  end
end

threads = []

10.times do
  threads << Thread.new { puts "[#{Thread.current}] #{Foo.foo}" }
end

threads.each(&:join)
```

This will sometimes set `foo` once, but sometimes multiple threads manage to set it and each sets it to a random number

Whereas this test case will only set `foo` once, even if multiple threads make it into the `synchronize` block:

```ruby
class Foo
  LOCK = Mutex.new

  def self.foo
    @foo || LOCK.synchronize do
      puts "Assigning value in #{Thread.current}"
      @foo ||= rand(100)
    end
  end
end

threads = []

10.times do
  threads << Thread.new { puts "[#{Thread.current}] #{Foo.foo}" }
end

threads.each(&:join)
```
</details>

Reply:
Aha! I wondered that but wasn't sure how to check it. Thanks!